.github/workflows/ci.yml

```diff
@@ -178,6 +178,94 @@ jobs:
 
+  MemoryTest:
+    runs-on: ubuntu-latest
+
+    #
+    # build matrix for containers
+    #
+    strategy:
+      #
+      # do not stop jobs automatically if any of the jobs fail
+      #
+      fail-fast: false
+
+      #
+      # matrix for type of checking
+      #
+      matrix:
+        checktype:
+          - glibc_debug
+          - sanitize_address
+          - sanitize_memory
+          - sanitize_thread
+          - sanitize_others
+          - valgrind
+
+    container:
+      image: ubuntu:22.04
+
+      options: "--privileged --cap-add SYS_ADMIN --device /dev/fuse"
+
+    env:
+      # [NOTE]
+      # Installation special environment variables for debian and ubuntu.
+      #
+      DEBIAN_FRONTEND:  noninteractive
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install packages
+        run: |
+          .github/workflows/linux-ci-helper.sh ubuntu:22.04
+
+      - name: Install clang
+        run: |
+          apt-get install -y clang
+          if [ "${{ matrix.checktype }}" = "valgrind" ]; then
+              apt-get install -y valgrind
+          fi
+
+      #
+      # Set CXX/CXXFLAGS and Variables for test
+      #
+      - name: Set variables
+        run: |
+          COMMON_CXXFLAGS='-g -O0 -Wno-cpp -DS3FS_PTHREAD_ERRORCHECK=1'
+          if [ "${{ matrix.checktype }}" = "glibc_debug" ]; then
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -D_GLIBCXX_DEBUG"  >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "sanitize_address" ]; then
+              echo 'CXX=clang++'                                   >> $GITHUB_ENV
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -fsanitize=address -fsanitize-address-use-after-scope" >> $GITHUB_ENV
+              echo 'ASAN_OPTIONS=detect_leaks=1,detect_stack_use_after_return=1'                       >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "sanitize_memory" ]; then
+              echo 'CXX=clang++'                                   >> $GITHUB_ENV
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -fsanitize=memory" >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "sanitize_thread" ]; then
+              echo 'CXX=clang++'                                   >> $GITHUB_ENV
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -fsanitize=thread" >> $GITHUB_ENV
+              echo 'TSAN_OPTIONS=halt_on_error=1'                  >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "sanitize_others" ]; then
+              echo 'CXX=clang++'                                   >> $GITHUB_ENV
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -fsanitize=undefined,implicit-conversion,local-bounds,unsigned-integer-overflow" >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "valgrind" ]; then
+              echo "CXXFLAGS=${COMMON_CXXFLAGS}"                   >> $GITHUB_ENV
+              echo 'VALGRIND=--leak-check=full'                    >> $GITHUB_ENV
```

Comment:
This might be too slow for CI.  We might add `-O1` which makes the tool a little less accurate but several times faster.

Reply:
Try changing to the `-O1` flag.